File: src/main/java/com/jss/osiris/modules/osiris/crm/service/CommunicationPreferenceService.java
package com.jss.osiris.modules.osiris.crm.service;

import com.jss.osiris.libs.exception.OsirisValidationException;
import com.jss.osiris.modules.osiris.crm.model.CommunicationPreference;

public interface CommunicationPreferenceService {

    public CommunicationPreference getCommunicationPreferenceByMail(String mail, String validationToken)
            throws OsirisValidationException;

    public CommunicationPreference subscribeToNewspaperNewsletter(String emailToSubscribe)
            throws OsirisValidationException;

    public CommunicationPreference unsubscribeToNewspaperNewsletter(String emailToSubscribe, String validationToken)
            throws OsirisValidationException;

    public CommunicationPreference subscribeToCorporateNewsletter(String emailToSubscribe)
            throws OsirisValidationException;

    public CommunicationPreference unsubscribeToCorporateNewsletter(String emailToSubscribe, String validationToken)
            throws OsirisValidationException;
}

File: client-myjss/src/app/modules/my-account/services/communication.preference.service.ts
import { HttpClient, HttpParams } from '@angular/common/http';
import { Injectable } from '@angular/core';
import { AppRestService } from '../../../libs/appRest.service';
import { CommunicationPreference } from '../model/CommunicationPreference';

@Injectable({
  providedIn: 'root'
})
export class CommunicationPreferencesService extends AppRestService<CommunicationPreference> {

  constructor(http: HttpClient) {
    super(http, "crm");
  }


  getCommunicationPreferenceByMail(mail: string, validationToken: string | null) {
    return this.get(new HttpParams().set("userMail", mail).set("validationToken", validationToken ? validationToken : "null"), "communication-preferences/communication-preference");
  }

  subscribeToNewspaperNewsletter(mail: string) {
    return this.get(new HttpParams().set("userMail", mail), "communication-preferences/subscribe-to-newspaper-newsletter");
  }

  unsubscribeToNewspaperNewsletter(mail: string, validationToken: string | null) {
    return this.get(new HttpParams().set("userMail", mail).set("validationToken", validationToken ? validationToken : "null"), "communication-preferences/unsubscribe-to-newspaper-newsletter");

  }

  subscribeToCorporateNewsletter(mail: string) {
    return this.get(new HttpParams().set("userMail", mail), "communication-preferences/subscribe-to-corporate-newsletter");

  }

  unsubscribeToCorporateNewsletter(mail: string, validationToken: string | null) {
    return this.get(new HttpParams().set("userMail", mail).set("validationToken", validationToken ? validationToken : "null"), "communication-preferences/unsubscribe-to-corporate-newsletter");

  }


}
